### PR TITLE
v3

### PR DESCRIPTION
**Pull request by robjtede**
- minor restyling of the site
- updated all examples to web v3
- plently of content updates

needs to be merged _after_ v3 release and extras crates have been updated

### CODE REVIEW

File: examples/application/Cargo.toml [hunk 1]
```diff
@@ -5,6 +5,6 @@ edition = "2018"
 workspace = "../"
 
 [dependencies]
-actix-web = "2.0"
-actix-rt = "1.0"
-actix-service = "1.0"
+actix-web = "3.0.0-beta.4"
```
Comment (Neopallium) [inline examples/application/Cargo.toml:8]: `actix-web = "3"`

File: examples/async-handlers/Cargo.toml [hunk 2]
```diff
@@ -4,7 +4,7 @@ version = "2.0.0"
 edition = "2018"
 
 [dependencies]
-actix-web = "2.0"
-actix-rt = "1.0"
+actix-web = "3.0.0-beta.4"
```
Comment (Neopallium) [inline examples/async-handlers/Cargo.toml:7]: Again use release version.

File: examples/databases/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+[package]
+name = "databases"
+version = "1.0.0"
+edition = "2018"
+
+[dependencies]
+actix-web = "3.0.0-beta.4"
```
Comment (Neopallium) [inline examples/databases/Cargo.toml:7]: Release version.

File: examples/easy-form-handling/Cargo.toml [hunk 4]
```diff
@@ -4,6 +4,6 @@ version = "1.0.0"
 edition = "2018"
 
 [dependencies]
-actix-web = "2.0"
-actix-rt = "1.0"
+actix-web = "3.0.0-beta.4"
```
Comment (Neopallium) [inline examples/easy-form-handling/Cargo.toml:7]: Release version.

File: examples/either/Cargo.toml [hunk 5]
```diff
@@ -4,5 +4,4 @@ version = "1.0.0"
 edition = "2018"
 
 [dependencies]
-actix-web = "1.0"
-futures = "0.1"
+actix-web = "3.0.0-beta.4"
```
Comment (Neopallium) [inline examples/either/Cargo.toml:7]: Release version.

File: examples/errors/Cargo.toml [hunk 6]
```diff
@@ -4,9 +4,9 @@ version = "1.0.0"
 edition = "2018"
 
 [dependencies]
-actix-web = "2.0.0"
-actix-rt = "1.0.0"
-env_logger = "0.7.1"
-log = "0.4.8"
-failure = "0.1.6"
-actix-http = "1.0.1"
+actix-web = "3.0.0-beta.4"
```
Comment (Neopallium) [inline examples/errors/Cargo.toml:7]: Release version.

File: examples/extractors/Cargo.toml [hunk 7]
```diff
@@ -4,7 +4,7 @@ version = "1.0.0"
 edition = "2018"
 
 [dependencies]
-actix-web = "2.0"
-actix-rt = "1.0"
+actix-web = "3.0.0-beta.4"
```
Comment (Neopallium) [inline examples/extractors/Cargo.toml:7]: Release version.

File: examples/extractors/src/json_two.rs [hunk 8]
```diff
@@ -12,28 +14,24 @@ async fn index(info: web::Json<Info>) -> impl Responder {
...
 async fn main() -> std::io::Result<()> {
-    use actix_web::{App, HttpServer};
-
     HttpServer::new(|| {
+        let json_config = web::Json::<Info>::configure(|cfg| {
+            cfg.limit(4096).error_handler(|err, _req| {
+                // create custom error response
+                error::InternalError::from_response(err, HttpResponse::Conflict().finish()).into()
+            })
+        });
```
Comment (Neopallium) [inline examples/extractors/src/json_two.rs:25]: Suggest:
```rust
        let json_config = JsonConfig::default().limit(4096)
            .error_handler(|err, _req| {
                // create custom error response
                error::InternalError::from_response(err, HttpResponse::Conflict().finish()).into()
            });
```

File: examples/flexible-responders/Cargo.toml [hunk 9]
```diff
@@ -4,6 +4,6 @@ version = "1.0.0"
 edition = "2018"
 
 [dependencies]
-actix-web = "2.0"
-actix-rt = "1.0"
+actix-web = "3.0.0-beta.4"
```
Comment (Neopallium) [inline examples/flexible-responders/Cargo.toml:7]: Release version.

File: examples/getting-started/Cargo.toml [hunk 10]
```diff
@@ -5,5 +5,5 @@ edition = "2018"
 workspace = "../"
 
 [dependencies]
-actix-web = "2.0"
+actix-web = "3.0.0-beta.4"
```
Comment (Neopallium) [inline examples/getting-started/Cargo.toml:8]: Release version.

File: examples/http2/Cargo.toml [hunk 11]
```diff
@@ -4,6 +4,5 @@ version = "1.0.0"
 edition = "2018"
 
 [dependencies]
-actix-web = { version = "2.0", features = ["openssl"] }
-actix-rt = "1.0"
+actix-web = { version = "3.0.0-beta.4", features = ["openssl"] }
```
Comment (Neopallium) [inline examples/http2/Cargo.toml:7]: Release version.

File: examples/main-example/Cargo.toml [hunk 12]
```diff
@@ -4,5 +4,5 @@ version = "1.0.0"
 edition = "2018"
 
 [dependencies]
-actix-web = "2.0"
-actix-rt = "1.0"
+actix-web = "3.0.0-beta.4"
```
Comment (Neopallium) [inline examples/main-example/Cargo.toml:7]: Release version.

File: examples/middleware/Cargo.toml [hunk 13]
```diff
@@ -4,9 +4,9 @@ version = "1.0.0"
 edition = "2018"
 
 [dependencies]
-actix-web = "2.0"
-actix-rt = "1.0"
-actix-service = "1.0"
+actix-web = "2"
```
Comment (Neopallium) [inline examples/middleware/Cargo.toml:7]: Release version "3" not "2".

File: examples/powerful-extractors/Cargo.toml [hunk 14]
```diff
@@ -4,6 +4,6 @@ version = "1.0.0"
 edition = "2018"
 
 [dependencies]
-actix-web = "2.0"
-actix-rt = "1.0"
+actix-web = "3.0.0-beta.4"
```
Comment (Neopallium) [inline examples/powerful-extractors/Cargo.toml:7]: Release version.

File: examples/request-handlers/Cargo.toml [hunk 15]
```diff
@@ -4,5 +4,5 @@ version = "1.0.0"
 edition = "2018"
 
 [dependencies]
-actix-web = "2.0"
-actix-rt = "1.0"
+actix-web = "3.0.0-beta.4"
```
Comment (Neopallium) [inline examples/request-handlers/Cargo.toml:7]: Release version.

File: examples/request-routing/Cargo.toml [hunk 16]
```diff
@@ -4,5 +4,5 @@ version = "1.0.0"
 edition = "2018"
 
 [dependencies]
-actix-web = "2.0"
-actix-rt = "1.0"
+actix-web = "3.0.0-beta.4"
```
Comment (Neopallium) [inline examples/request-routing/Cargo.toml:7]: Release version.

File: examples/requests/Cargo.toml [hunk 17]
```diff
@@ -6,8 +6,8 @@ edition = "2018"
 [dependencies]
 serde = "1.0"
 serde_json = "1.0"
-actix-web = "2.0"
-actix-rt = "1.0"
+actix-web = "3.0.0-beta.4"
```
Comment (Neopallium) [inline examples/requests/Cargo.toml:9]: Release version.

File: examples/responder-trait/Cargo.toml [hunk 18]
```diff
@@ -4,8 +4,8 @@ version = "1.0.0"
 edition = "2018"
 
 [dependencies]
-actix-web = "2.0"
-actix-rt = "1.0"
+actix-web = "3.0.0-beta.4"
```
Comment (Neopallium) [inline examples/responder-trait/Cargo.toml:7]: Release version.

File: examples/responses/Cargo.toml [hunk 19]
```diff
@@ -4,8 +4,8 @@ version = "1.0.0"
 edition = "2018"
 
 [dependencies]
-actix-web = "2.0"
-actix-rt = "1.0"
+actix-web = "3.0.0-beta.4"
```
Comment (Neopallium) [inline examples/responses/Cargo.toml:7]: Release version.

File: examples/server/Cargo.toml [hunk 20]
```diff
@@ -5,8 +5,6 @@ workspace = "../"
 edition = "2018"
 
 [dependencies]
-actix-rt = "1.0"
-actix-web = { version = "2.0", features = ["openssl"] }
-futures = "0.3.1"
+actix-web = { version = "3.0.0-beta.4", features = ["openssl"] }
```
Comment (Neopallium) [inline examples/server/Cargo.toml:8]: Release version.

File: examples/static-files/Cargo.toml [hunk 21]
```diff
@@ -4,7 +4,7 @@ version = "1.0.0"
 edition = "2018"
 
 [dependencies]
-actix-web = "2.0"
-actix-rt = "1.0"
-actix-files = "0.2"
-mime = "*"
+actix-web = "3.0.0-beta.4"
```
Comment (Neopallium) [inline examples/static-files/Cargo.toml:7]: Release version.

File: examples/testing/Cargo.toml [hunk 22]
```diff
@@ -4,8 +4,8 @@ version = "1.0.0"
 edition = "2018"
 
 [dependencies]
-actix-web = "2.0"
-actix-rt = "1.0"
+actix-web = "3.0.0-beta.4"
```
Comment (Neopallium) [inline examples/testing/Cargo.toml:7]: Release version.

File: examples/url-dispatch/Cargo.toml [hunk 23]
```diff
@@ -6,8 +6,8 @@ workspace = "../"
 
 [dependencies]
 actix = "0.9"
-actix-rt = "1.0"
-actix-web = "2.0"
+actix-rt = "1"
+actix-web = "3.0.0-beta.4"
```
Comment (Neopallium) [inline examples/url-dispatch/Cargo.toml:10]: Release version.

File: examples/websockets/Cargo.toml [hunk 24]
```diff
@@ -4,7 +4,7 @@ version = "1.0.0"
...
 
 [dependencies]
-actix = "0.9"
-actix-rt = "1.0"
-actix-web = "2.0"
-actix-web-actors = "2.0"
+actix = "0.10.0-alpha.3"
+actix-rt = "1"
+actix-web = "3.0.0-beta.4"
+actix-web-actors = "3.0.0-beta.2"
```
Comment (Neopallium) [inline examples/websockets/Cargo.toml:10]: Release versions.

**Comment by Neopallium** [comment]
Mostly just some `beta` crates still used and one suggested `JsonConfig` change.

File: README.md [hunk 25]
```diff
@@ -35,5 +38,11 @@ mmdc -i connection_overview.mmd -o connection_overview.svg
 
 # License
 
-Pretty murky.  Right now a massive clone of the tokio website.  Will get this
+Pretty murky. Right now a massive clone of the tokio website. Will get this
```
Comment (popzxc) [inline README.md:41]: Is this statement still valid? I think it's safe to change to creative commons.

Comment (robjtede) [reply README.md:41]: tbh i'm not sure if its still valid, can't remember what the tokio site used to look like. Something to sort out later, ideally with a larger restyling of the site.

**Pull request merged**